cogs/remind_me.py

```diff
@@ -139,12 +138,17 @@ async def autocomplete_get_delays(ctx: TeXBotAutocompleteContext) -> set[str]:
 
                 slice_size: int
                 for slice_size in range(1, len(formatted_time_choice) + 1):
-                    if match.group("ctx_time_choice").casefold() == formatted_time_choice[:slice_size]:  # noqa: E501
+                    if (
+                        match.group("ctx_time_choice").casefold()
+                        == formatted_time_choice[:slice_size]
+                    ):
                         delay_choices.add(formatted_time_choice[slice_size:])
 
         elif match := re.match(r"\A(?P<date>\d{1,2}) ?[/\-.] ?\Z", ctx.value):
             if 1 <= int(match.group("date")) <= 31:
-                FORMATTED_DAY_AND_JOINER_DATE_CHOICES: Final[Iterator[tuple[int, int, str]]] = itertools.product(  # noqa: E501
+                FORMATTED_DAY_AND_JOINER_DATE_CHOICES: Final[
+                    Iterator[tuple[int, int, str]]
+                ] = itertools.product(
```

Comment:
Type annotations should be kept on the same line (even if that extends the max line length). The line can be reduced in this case by adding the `itertools.product` function call to the line below:

```python
FORMATTED_DAY_AND_JOINER_DATE_CHOICES: Final[Iterator[tuple[int, int, str]]] = (  # noqa: E501
    itertools.product(
        range(1, 12),
        range(current_year, current_year + 40),
        ("/", " / ", "-", " - ", ".", " . "),
    )
)
```